### PR TITLE
Add b3 metadata in grpc

### PR DESCRIPTION
**Pull request by henrich-m**
**What does this PR do?**
It adds the support of B3 Metadata injection in gRPC calls.

**Motivation**
The B3 Standard is adopted in almost all observability frameworks, including dd-trace, it already does the injection and extraction in HTTP calls, but not in GRPC calls, we have 2 backends services that communicate with each other through gRPC and I can't see the traces correlated because one of them is written in Golang and utilizes Open Telemetry, that's why we need the B3 also in Metadata. 

**How to test the change?**
Unit tests should cover the changes. :)

### CODE REVIEW

**Review by marcotc — COMMENTED** [review]
👋 @henrich-m, this PR is super flushed out, thank you! I left a minor comment.

If I can: could you add a new configuration to the gRPC integration: `option :distributed_tracing, default: true` (at `lib/datadog/tracing/contrib/grpc/configuration/settings.rb`)? This allows anyone to disable header propagation, in case gPRC is used strictly to contact 3rd parties: in that case, you might not want to expose internal headers to them. You can see most of other HTTP integrations implement such option.

File: lib/datadog/tracing/distributed/headers/ext.rb [hunk 1]
```diff
@@ -24,6 +24,12 @@ module Ext
           GRPC_METADATA_PARENT_ID = 'x-datadog-parent-id'.freeze
           GRPC_METADATA_SAMPLING_PRIORITY = 'x-datadog-sampling-priority'.freeze
           GRPC_METADATA_ORIGIN = 'x-datadog-origin'.freeze
+
+          # B3 gRPC metadata used for distributed tracing
+          B3_METADATA_TRACE_ID = 'x-b3-traceid'.freeze
+          B3_METADATA_SPAN_ID = 'x-b3-spanid'.freeze
+          B3_METADATA_SAMPLED = 'x-b3-sampled'.freeze
+          B3_METADATA_SINGLE = 'b3'.freeze
```
[the remark this answers is not in the document]
Comment (henrich-m) [reply lib/datadog/tracing/distributed/headers/ext.rb:32]: I was just following the same approach adopted for Datadod headers, I will refactor them :D

**Review by marcotc — APPROVED** [review]
Thank you, @henrich-m! 🙇

**Comment by ivoanjo** [comment]
👋 @henrich-m this change has been released in [v1.2.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.2.0).
Thanks a bunch for the awesome contribution :)

**Pull request merged**